File: frontend/src/helpers/regex.ts
const NameFieldWithSpecialCharacters = `[-*]?\\w+`
const valueInnerParentheses = `\\([^)]+\\)`

export const fieldRegex = new RegExp(`${NameFieldWithSpecialCharacters}:(${valueInnerParentheses}|\\S+)`)
export const specialCharactersRegex = /[<>'";&|$\\[\\]{}#%=]/

export const sanitizeInput = (query:string) =>{
  return query.replace(specialCharactersRegex,"")
}

File: api/helpers/regex.go
package helpers

import "fmt"

var ValueInnerParentheses = `\([^)]+\)`
var NameFieldWithSpecialCharacters = `[-*]?\w+`
var FieldRegex = fmt.Sprintf(`%v:(%v|\S+)`, NameFieldWithSpecialCharacters, ValueInnerParentheses)
